node/node.go

```diff
@@ -700,9 +703,15 @@ func NewNode(config *cfg.Config,
 		}
 	}
 
-	// Determine whether we should do fast sync. This must happen after the handshake, since the
-	// app may modify the validator set, specifying ourself as the only validator.
-	fastSync := config.FastSyncMode && !onlyValidatorIsUs(state, pubKey)
+	// Determine whether we should do fast sync.
+	//
+	// NOTE: This must happen after the handshake, since the app may modify the
+	// validator set, specifying ourself as the only validator.
+	//
+	// If the validator has 1/3+ of the voting power, no new blocks were minted
+	// since 2/3+ of the voting power is required, hence we can safely skip fast
+	// and state sync.
+	fastSync := config.FastSyncMode && !validatorHasMoreThan13Power(state, pubKey)
```

Comment:
There are cases where this may be incorrect. For example, if we're starting a blank node which had 1/3+ voting power in genesis but the chain is now at a higher height where the validator is no longer part of the validator set, then it will be impossible to use fast sync to resync that node.

Reply:
> but the chain is now at a higher height

but that's impossible

Reply:
Not if the node has e.g. a hard drive failure or database corruption, and has to start over from scratch.

Reply:
right, that is possible.